frontend/src/app/main/ui/workspace/context_menu.cljs

```diff
@@ -435,17 +435,19 @@
 (mf/defc context-menu-component
   [{:keys [shapes]}]
   (let [single?             (= (count shapes) 1)
+        components-v2       (features/use-feature :components-v2)
 
         has-component?      (some true? (map #(contains? % :component-id) shapes))
         is-component?       (and single? (-> shapes first :component-id some?))
         is-non-root?        (and single? (ctk/in-component-copy-not-root? (first shapes)))
+        is-touched?         (and single? (seq (:touched (first shapes))))
```

Comment:
This does not work. Update and reset should appear if the head shape **or any of the child shapes** are touched.

Reply:
Absolutely true! Fixed!